openupgrade_scripts/scripts/account/17.0.1.2/post-migration.py

```diff
@@ -558,6 +558,104 @@ def _rename_coa_elements_xmlids(env):
         )
 
 
+def _assign_journal_xmlids(env):
+    """
+    Starting in Odoo 17, Odoo expects journals to get external id
+    like account.1_sale for sale journal on company 1
+    When updating chart of accounts, Odoo tries to resolve missing external ids
+    using code or name from journal, if none match, a new journal would be created
+    We therefore go one step further here to assign by default journals by type only
+    The below code has been derived from Odoo SA since it is inside _pre_reload_data
+    function and cannot be executed without surrounding function lines
+    https://github.com/odoo/odoo/blob/17.0/addons/account/models/chart_template.py#L257
+    Therefore Odoo SA copyright applies to this part of code
+    """
+    Template = env["account.chart.template"]
+    env.cr.execute(
+        """SELECT id AS company_id, chart_template
+            FROM res_company
+            WHERE chart_template IS NOT NULL
+        """,
+    )
+    for company_id, chart_template in env.cr.fetchall():
+        company = env["res.company"].browse(company_id)
+        template_data = Template._get_chart_template_data(chart_template)
```

Comment:
It seems this script has to be an end-migration.

Reply:
I did not get that error while testing locally, but yes it makes sense. I just fixed that !
Thanks !